migration-engine/core/src/api/rpc.rs

```diff
@@ -7,7 +7,7 @@ use std::sync::Arc;
 /// A JSON-RPC ready migration API.
 pub struct RpcApi {
     io_handler: jsonrpc_core::IoHandler<()>,
-    executor: Arc<dyn GenericApi>,
+    executor: Arc<Box<dyn GenericApi>>,
```

Comment:
?

Reply:
We only need an `Arc` in that one place, in the rest of the calls it can be a `Box`, so I prefer to return a `Box` from `migration_core::migration_api()`.